types/pouchdb-core/index.d.ts

```diff
@@ -55,6 +55,15 @@ declare namespace PouchDB {
             rev: RevisionId;
             status: Availability;
         }
+        interface RevisionDiffList {
+            [documentId : string]: string[];
+        }
+        interface RevisionDiff {
+            [change : string]: string[];
```

Comment:
Can you try using literal join types? Like so:
```ts
type RevisionDiffStatus = 'missing' | 'possible_ancestors'
```

That, imo, should trick typescript into believing that type is a subtype of String (which it is), and thus allow it.